doc/whats_new.rst

```diff
@@ -44,6 +44,7 @@ Detailed list of changes
 - Add :meth:`mne_bids.BIDSPath.find_matching_sidecar` to find the sidecar file associated with a given file path by `Eric Larson`_ (:gh:`1093`)
 - When writing data via :func:`~mne_bids.write_raw_bids`, it is now possible to specify a custom mapping of :class:`mne.Annotations` descriptions to event codes via the ``event_id`` parameter. Previously, passing this parameter would always require to also pass ``events``, and using a custom event code mapping for annotations was impossible, by `Richard Höchenberger`_ (:gh:`1084`)
 - Improve error message when :obj:`~mne_bids.BIDSPath.fpath` cannot be uniquely resolved by `Eric Larson`_ (:gh:`1097`)
+- Add :func:`mne_bids.find_matching_paths` to get all bids paths that match the requirements by :newcontrib:`Moritz Gerster` (:gh:`1103`)
```

Comment:
Is this correct @hoechenberger? It is not described how to add lines in the [mne_bids contribution guide](https://github.com/mne-tools/mne-bids/blob/main/CONTRIBUTING.md) so I followed the [mne contributing guide](https://mne.tools/stable/install/contributing.html). 

```
The following authors contributed for the first time. Thank you so much! 🤩

* ...
```
Here I didn't add my name because I guess this works automatically?

Reply:
> I guess this works automatically

unfortunately not :-)

> I followed the [mne contributing guide](https://mne.tools/stable/install/contributing.html).

it's a bit different here. You just add your name to the list you quoted above, and we don't have the sphinx role ":new_contrib:", so you can just follow the style of previous changelog entries.

Reply:
Nevermind, I checked the old versions of `whats_new.rst` and understood that you don't have `:newcontrib:` as in mne.

Reply:
> if you want the function to be available as `mne_bids.find_matching_paths`, for example, you'd need to make additions here:
> 
> https://github.com/mne-tools/mne-bids/blob/b97ce0b336f2c635e45d0436eb7b433c33b75a16/mne_bids/__init__.py#L4-L18
> 
> for example look at `BIDSPath` in that file --> if it weren't where it is, you would always need to do `mne_bids.path.BIDSPath`.
> 
> And for the API docs, you need to add it here (provided you followed the above):
> 
> https://github.com/mne-tools/mne-bids/blob/b97ce0b336f2c635e45d0436eb7b433c33b75a16/doc/api.rst?plain=1#L19-L22

Thanks @sappelhoff! Added with [1bc097e](https://github.com/mne-tools/mne-bids/pull/1103/commits/1bc097e482c874b7ee5b8da06a5df95759979541).